src/components/SignInPopup/SignInPopup.js

```diff
@@ -0,0 +1,138 @@
+import "./SignInPopup.css";
+import { useState } from "react";
+import PopupWithForm from "../PopupWithForm/PopupWithForm";
+import * as auth from "../../utils/auth";
+import validation from "../../utils/validate";
+
+function SignInPopup({ isOpen, onClose, openSignUpPopup, handleLogin }) {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  let activeSubmit=false;
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  if(email && password && emailError ==="" && passwordError===""){
+    activeSubmit=true;
+  }
+
+
+  function onEmailChange(e) {
+    const currentEmailInput = e.target.value;
+    setEmail(currentEmailInput);
+    if (
+      currentEmailInput.length > 0 &&
+      !validation(currentEmailInput, "email")
+    ) {
+      setEmailError("Provide a valid email address");
+    } else if (
+      (currentEmailInput.length > 0 &&
+        validation(currentEmailInput, "password")) ||
+      currentEmailInput.length === 0
+    ) {
+      setEmailError("");
+    }
+  }
+
+  function onPasswordChange(e) {
+    const currentPasswordInput = e.target.value;
+    setPassword(currentPasswordInput);
+    if (
+      currentPasswordInput.length > 0 &&
+      !validation(currentPasswordInput, "password")
+    ) {
+      setPasswordError("Password must consist of at least 8 characters");
+    } else if (
+      (currentPasswordInput.length > 0 &&
+        validation(currentPasswordInput, "password")) ||
+      currentPasswordInput.length === 0
+    ) {
+      setPasswordError("");
+    }
+  }
+
+  
+
+  function resetForm() {
+    setEmail("");
+    setPassword("");
+  }
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    if (!email || !password) {
+      return;
+    }
+    auth
+      .authorize(password, email)
```

Comment:
**Might be better**

it's better to make api requests only in `App` because `Login` and `Register` should not know how to do it. They should be independent from such logics